submissions/morenko/oop-exercise/index.js

```diff
@@ -0,0 +1,86 @@
+// создадим родительский класс World
+class WorldInhabitant {
+    // обьявляем поля (свойства)
+    
+    // объявляем метод конструктора (ожидаем на входе значения)
+    constructor(name, species, gender, saying){
+        // инициализация полей
+        this.name = name;
+        this.species = species;
+        this.gender = gender;
+        this.saying = saying;
+    }
+  // объявление метода toString()
+  // toString() возвращает строку, представляющую объект
+  // обращение к собственному атрибуту через this
+    toString() {
+       return `${this.name}; ${this.species}; ${this.gender}; ${this.saying}; `;
+    }
+  }
+  
+  // создадим два класса наследующие от родительского часть свойств
+  class Human extends WorldInhabitant {
+    constructor(name, species, gender, legs, hands, saying) {
+       super(name, species, gender, saying); //наследуем поля от родителя, и добавляем еще два поля
+       this.legs = legs;
+       this.hands = hands;
+    } 
+    
+    toString() {
+      return super.toString() + `${this.legs}; ${this.hands}; `
+    }
+  }
+  
+  class Animal extends WorldInhabitant {
+    // наследуем четыре свойства от родителя и добавляем лапы для животных
+    constructor(name, species, gender, paws, saying) {
```

Comment:
Same here, find a way to make it so you don't pass species anywhere, let classes handle it.

Reply:
And animals in this exercise always have 4 legs.